tree/dataframe/inc/ROOT/RDF/RLoopManager.hxx

```diff
@@ -131,6 +131,8 @@ class RLoopManager : public RNodeBase {
    std::vector<ROOT::RDF::SampleCallback_t> fSampleCallbacks;
    RDFInternal::RNewSampleNotifier fNewSampleNotifier;
    std::vector<ROOT::RDF::RSampleInfo> fSampleInfos;
+   ROOT::Internal::TreeUtils::RFriendInfo fFriendInfo{};
+   std::vector<std::unique_ptr<TTree>> fFriends; ///< Friends of the tree/chain, if present
```

Comment:
I don't understand why we need all these extra data members, especially the last one. The data members that belong to the spec should be removed, instead RLoopManager holds a RDataSetSpec as we said. I'm unsure about the vector of pointers. I am scared that we need it only because TChain doesn't keep alive the friends even after you attach them to it

Reply:
I realize now that `fFriendInfo` is not supposed to be there. It is true that I can be getting that from the `fSampleInfos` - I am applying this small change in a bit.

Regarding the vector of pointers - yes, it is exactly lifetime issues. If I don't have this vector => I cannot add the friends inside of the RLoopManager ctor. The alternative to keep them only inside `RunTreeReader` and `RunTreeProcessorMT`.

Reply:
Now I recall why I added `fFriendInfo` in the RLoopManager. I saw that we still added:
```cpp
   ULong64_t fStartEntry{0};
   ULong64_t fEndEntry{std::numeric_limits<ULong64_t>::max()};
``` 

I support the argument of redundancy, so I think we do not need `fStartEntry` and `fEndEntry` as they are already present in the RDatasetSpec, right?